src/deferred.js

```diff
@@ -321,29 +319,29 @@ jQuery.extend( {
 					}
 				};
 			},
-			progressValues, progressContexts, resolveContexts;
+			resolveContexts;
 
-		// Add listeners to Deferred subordinates; treat others as resolved
+		// Add listeners to promise-like subordinates; treat others as resolved
 		if ( length > 0 ) {
-			progressValues = new Array( length );
-			progressContexts = new Array( length );
 			resolveContexts = new Array( length );
 			for ( ; i < length; i++ ) {
```

Comment:
It would, although I'm not sure that would be observable? Since we use Deferred with `.done()` it will cause the resolveValues to be filled in a different order if it happens synchronously, but we don't do anything with it externally until all the subordinates resolve.